web/tests/proportion-field-directive-spec.js

```diff
@@ -96,34 +96,44 @@ describe('proportion-field directive', function () {
 
     describe('setting model', function () {
         beforeEach(function () {
-            this.total = 100;
-            spyOn(this.proportionField, 'calculateTotal').and.returnValue(this.total);
-            spyOn(this.proportionField, 'addPercentages');
             this.scope.options = this.options;
             this.scope.model = this.model;
             this.compileDirective();
             this.isolated = this.elm.isolateScope()
             this.scope.$digest();
         });
 
-        it('should call proportionField.calculateTotal and set scope.total', function () {
-            expect(this.proportionField.calculateTotal).toHaveBeenCalledWith(this.model)
-            expect(this.isolated.total).toBe(this.total);
+        it('should create an empty opbject if the model is null', function () {
+            this.scope.model = null;
+            this.scope.$digest();
+            expect(this.isolated.model).toEqual({});
         });
 
-        it('should call addPercentages', function () {
-            expect(this.proportionField.addPercentages).toHaveBeenCalledWith(
-                this.isolated.fields,
-                this.model,
-                this.total
-            )
+        it('should call proportionField.calculateTotal and set scope.total', function () {
+            spyOn(this.proportionField, 'calculateTotal').and.returnValue(this.total);
```

Comment:
I can't see where this.total gets set - Possibly it is `undefiend` maybe it should be a variable local to this function?